posthog/temporal/workflows/batch_exports.py

```diff
@@ -634,6 +635,14 @@ async def execute_batch_export_insert_activity(
         initial_retry_interval_seconds: When retrying, seconds until the first retry.
         maximum_retry_interval_seconds: Maximum interval in seconds between retries.
     """
+
+    def capture_exception_with_metadata(e: BaseException):
```

Comment:
nit: catching `BaseException` is usually bad (or very low-level) as it includes stuff like `KeyboardInterruptedError`. `Exception` should be more than okay (and it's what we catch).

Reply:
`mypy` actually yelled at me about this, but it may have just been mad that I was possibly passing in `None` (it referred to `Optional[BaseException]`) which I've fixed, so I can try again.

That said, technically I'm not changing what we catch at all, here, those `except` branches below are the same.

Reply:
Yeah, type hints are merely hints. This was a very nit-picky comment.

Reply:
Yeah, it really wants me to use this:

```
posthog/temporal/workflows/batch_exports.py:667: error: Argument 1 to "capture_exception_with_metadata" has incompatible type "BaseException"; expected "Exception"
```

Reply:
Thank you for looking into this! We can figure it out later, I was being nit-picky anyways.